pkg/volume/csi/csi_mounter.go

```diff
@@ -365,7 +365,9 @@ func (c *csiMountMgr) TearDownAt(dir string) error {
 		return errors.New(log("mounter.TearDownAt failed: %v", err))
 	}
 
-	// clean mount point dir
+	// Deprecation: Removal of this mount directory MUST be done by
+	// the CSI plugin according to the spec. This will no longer be done
+	// directly as part of TearDown by the kubelet in the future.
 	if err := removeMountDir(c.plugin, dir); err != nil {
```

Comment:
sorry about keep going back and forth on this but - we should be explicit in comment as well that we are only deprecating support for removing NodePublish target_path, removal of parent directories are still a responsibility of kubelet.

So something like:

Deprecation: Removal of target_path provided in NodePublish RPC call (in this case location `dir`)
MUST be done by the CSI plugin according to the spec. 
This will no longer be done directly as part of TearDown by the kubelet in the future.Kubelet will only be
responsible for removal of json data files it creates and parent directories.

Reply:
All good, thanks for the feedback :+1: